Chunks/Chunk.cs

```diff
@@ -135,9 +136,19 @@ public void CreateCollisionSurface(Simulation simulation, BufferPool bufferPool)
             _shape));
     }
 
+    public void Dispose(Simulation simulation, BufferPool bufferPool)
+    {
+        if (_shape.Exists)
```

Comment:
I'm not sure if this will actually check if the shape is present in the simulation if that's the intention

Reply:
Me neither but this part was throwing exceptions and not its not so i'm guessing that this works